tests/components/goodwe/test_config_flow.py

```diff
@@ -0,0 +1,108 @@
+"""Test the Goodwe config flow."""
+from unittest.mock import AsyncMock, patch
+
+from goodwe import InverterError
+import pytest
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.goodwe.const import (
+    CONF_MODEL_FAMILY,
+    CONF_NETWORK_RETRIES,
+    CONF_NETWORK_TIMEOUT,
+    DEFAULT_NAME,
+    DOMAIN,
+)
+from homeassistant.const import CONF_HOST, CONF_SCAN_INTERVAL
+
+from tests.common import MockConfigEntry
+
+TEST_HOST = "1.2.3.4"
+TEST_SERIAL = "123456789"
+
+
+def mock_inverter():
+    """Get a mock object of the inverter."""
+    goodwe_inverter = AsyncMock()
+    goodwe_inverter.serial_number = TEST_SERIAL
+    return goodwe_inverter
+
+
+@pytest.fixture(name="goodwe_connect", autouse=True)
+def goodwe_connect_fixture():
+    """Mock motion blinds connection and entry setup."""
+    with patch(
+        "homeassistant.components.goodwe.config_flow.connect",
+        return_value=mock_inverter(),
+    ), patch("homeassistant.components.goodwe.async_setup_entry", return_value=True):
+        yield
+
+
+async def test_config_flow_manual_host_success(hass):
```

Comment:
Make sure to test already configured entry with the same unique id. Add a MockConfigEntry with the same unique id first.

Reply:
This still remains to be done.

Reply:
Done.